sdk/keyvault/keyvault-keys/src/cryptographyClient.ts

```diff
@@ -590,11 +610,15 @@ export class CryptographyClient {
   ): Promise<CryptographyProvider> {
     if (!this.providers) {
       const keyMaterial = await this.getKeyMaterial(options);
+      this.providers = [];
+
       // Add local crypto providers as needed
-      this.providers = [
-        new RsaCryptographyProvider(keyMaterial),
-        new AesCryptographyProvider(keyMaterial),
-      ];
+      if (keyMaterial) {
+        this.providers.push(
+          new RsaCryptographyProvider(keyMaterial),
+          new AesCryptographyProvider(keyMaterial)
```

Comment:
I'm guessing you don't have ECDsa support, but what happens in that case? Do the providers get queried for what `kty`s they support?

Reply:
Each provider has an `isSupported` operation taking in an algorithm/operation pair, for example:

https://github.com/Azure/azure-sdk-for-js/blob/4091d3192e073e6de56f098cdc61c28273456248/sdk/keyvault/keyvault-keys/src/cryptography/rsaCryptographyProvider.ts#L41-L46

If that returns false then we fall back to the next provider in the list. So if no local provider supports the algorithm and operation, it will fall back all the way to the remote provider.